spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java

```diff
@@ -60,41 +74,131 @@ public void tearDown() {
 		context.refresh();
-		this.listener = context.getBean(TaskLifecycleListener.class);
-		TestVerifierUtils.verifyLogEntryExists(mockAppender,
-				"Creating: TaskExecution{executionId=" +
-						listener.getTaskExecution().getExecutionId());
-		assertEquals("Create should report that exit code is zero",
-				0, listener.getTaskExecution().getExitCode());
+		verifyTaskExecution(0, false, 0, null);
+	}
 
+	@Test
+	public void testTaskCreateWithArgs() {
+		context.register(ArgsConfiguration.class);
+		context.refresh();
+		verifyTaskExecution(2, false, 0, null);
 	}
 
 	@Test
 	public void testTaskUpdate() {
 		context.refresh();
-		final Appender mockAppender = TestVerifierUtils.getMockAppender();
-		this.listener = context.getBean(TaskLifecycleListener.class);
-		this.listener.onApplicationEvent(new ContextClosedEvent(context));
-		TestVerifierUtils.verifyLogEntryExists(mockAppender,
-				"Updating: TaskExecution{executionId=" +
-						listener.getTaskExecution().getExecutionId());
-		assertEquals("Update should report that exit code is zero",
-				0, listener.getTaskExecution().getExitCode());
+
+		context.publishEvent(new ContextClosedEvent(context));
+
+		verifyTaskExecution(0, true, 0, null);
 	}
 
 	@Test
 	public void testTaskFailedUpdate() {
 		context.refresh();
-		final Appender mockAppender = TestVerifierUtils.getMockAppender();
-		this.listener = context.getBean(TaskLifecycleListener.class);
-		listener.onApplicationEvent(new ApplicationFailedEvent(new SpringApplication(), new String[]{}, context, new RuntimeException("This was expected")));
-		TestVerifierUtils.verifyLogEntryExists(mockAppender,
-				"Updating: TaskExecution{executionId=" +
-						listener.getTaskExecution().getExecutionId());
-		assertEquals("Update should report that exit code is one",
-				1, listener.getTaskExecution().getExitCode());
-		assertTrue("Stack trace missing from exit message", listener.getTaskExecution().getExitMessage().startsWith("java.lang.RuntimeException: This was expected"));
+		RuntimeException exception = new RuntimeException("This was expected");
+		context.publishEvent(new ApplicationFailedEvent(new SpringApplication(), new String[0], context, exception));
+
+		verifyTaskExecution(0, true, 1, exception);
+	}
+
+	private static String stackTraceToString(Throwable exception) {
+		StringWriter writer = new StringWriter();
+		PrintWriter printWriter = new PrintWriter(writer);
+
+		exception.printStackTrace(printWriter);
+
+		return writer.toString();
+	}
+
+	private void verifyTaskExecution(int numberOfParams, boolean update, int exitCode, Throwable exception) {
+		this.taskExplorer = context.getBean(TaskExplorer.class);
+
+		Sort sort = new Sort("id");
+
+		PageRequest request = new PageRequest(0, Integer.MAX_VALUE, sort);
+
+		Page<TaskExecution> taskExecutionsByName = this.taskExplorer.findTaskExecutionsByName("testTask",
+				request);
+
+		TaskExecution taskExecution = taskExecutionsByName.iterator().next();
```

Comment:
Probably want to add an assertTrue for taskExecutionsByName.iterator().hasNext first before executing the next.